weld/partial_types.rs

```diff
@@ -10,6 +10,7 @@ use super::error::*;
 pub enum PartialType {
     Unknown,
     Scalar(ScalarKind),
+    Vectorized(ScalarKind),
     Vector(Box<PartialType>),
```

Comment:
Maybe one idea is to switch `Vectorized` to `Simd`. Or we can change `Vectorized` to `Vector` and `Vector` to `Seq` or `List` or something.